src/Lib.P2P.Tests/CidTest.cs

```diff
@@ -0,0 +1,485 @@
+            Assert.AreEqual(0, cid.Version);
+            Assert.AreEqual("dag-pb", cid.ContentType);
+            Assert.AreEqual("base58btc", cid.Encoding);
+            Assert.AreEqual(hash, cid.Encode());
+        }
+
+        [TestMethod]
+        public void Decode_V0_Invalid()
+        {
+            var hash = "QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCVVEcRTWJBmLgR39?";
+            Assert.ThrowsException<FormatException>(() => Cid.Decode(hash));
+        }
+
+        [TestMethod]
+        public void Decode_Invalid_Version()
+        {
+            var cid = new Cid
+            {
+                Version = 32767,
+                ContentType = "raw",
+                Encoding = "base58btc",
+                Hash = "QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4"
+            };
+            var s = cid.Encode();
+            Assert.ThrowsException<FormatException>(() => Cid.Decode(s));
+        }
+
+        [TestMethod]
+        public void Decode_V1()
+        {
+            var id = "zb2rhj7crUKTQYRGCRATFaQ6YFLTde2YzdqbbhAASkL9uRDXn";
+            var hash = "QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4";
+            var cid = Cid.Decode(id);
+            Assert.AreEqual(1, cid.Version);
+            Assert.AreEqual("base58btc", cid.Encoding);
+            Assert.AreEqual("raw", cid.ContentType);
+            Assert.AreEqual(hash, cid.Hash);
+        }
+
+        [TestMethod]
+        public void Decode_V1_Unknown_ContentType()
+        {
+            var id = "zJAFhtPN28kqMxDkZawWCCL52BzaiymqFgX3LA7XzkNRMNAN1T1J";
+            var hash = "QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4";
+            var cid = Cid.Decode(id);
+            Assert.AreEqual(1, cid.Version);
+            Assert.AreEqual("base58btc", cid.Encoding);
+            Assert.AreEqual("codec-32767", cid.ContentType);
+            Assert.AreEqual(hash, cid.Hash);
+        }
+
+        [TestMethod]
+        public void Decode_V1_Invalid_MultiBase_String()
+        {
+            var id = "zb2rhj7crUKTQYRGCRATFaQ6YFLTde2YzdqbbhAASkL9uRDX?";
+            Assert.ThrowsException<FormatException>(() => Cid.Decode(id));
+        }
+
+        [TestMethod]
+        public void Decode_V1_Invalid_MultiBase_Code()
+        {
+            var id = "?";
+            Assert.ThrowsException<FormatException>(() => Cid.Decode(id));
+        }
+
+        [TestMethod]
+        public void Value_Equality()
+        {
+            var a0 = Cid.Decode("zb2rhj7crUKTQYRGCRATFaQ6YFLTde2YzdqbbhAASkL9uRDXn");
+            var a1 = Cid.Decode("zb2rhj7crUKTQYRGCRATFaQ6YFLTde2YzdqbbhAASkL9uRDXn");
+            var b = Cid.Decode("QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L5");
+            Cid c = null;
+            Cid d = null;
+
+            Assert.IsTrue(c == d);
+            Assert.IsFalse(c == b);
+            Assert.IsFalse(b == c);
+
+            Assert.IsFalse(c != d);
+            Assert.IsTrue(c != b);
+            Assert.IsTrue(b != c);
+
+#pragma warning disable 1718
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)